pkg/controller/registry/resolver/cache/cache.go

```diff
@@ -139,7 +139,7 @@ func (c *NamespacedOperatorCache) Error() error {
 		err := snapshot.err
 		snapshot.m.RUnlock()
 		if err != nil {
-			errs = append(errs, fmt.Errorf("failed to populate resolver cache from source %v: %w", key.String(), err))
+			errs = append(errs, fmt.Errorf("error using catalog %s (in namespace %s): %w", key.Name, key.Namespace, err))
```

Comment:
This is for handling a deleted catalog, right? Presumably the deletion happened intentionally. At this point in the code we simply can't find it. Would it be better to Warn instead?

Reply:
No, if a transient network failure prevents the catalog operator from reaching a particular registry server, ignoring the error means also ignoring that we weren't able to fetch all of a catalog's content. That could result in different operators being installed.

Reply:
wasn't talking about a logic change. just the log level, Error -> Warn. It's something we're going to retry or that possibly a catalog was intentionally removed. Was just thinking Warn might be more appropriate for the situation.

Reply:
>This is for handling a deleted catalog, right?

@bentito not really. This is the resolver saying "I cannot use this catalog, for some reason or the other". [This test case](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/pkg/controller/registry/resolver/step_resolver_test.go#L1203-L1225) demonstrates that well. The catalog cannot be used in that case because "csv" (tbh on the first read through I don't exactly get what that means 😄, probably a needs to be worked on a bit to make it more clear, but the point is because of "3EntryReplacementChain" and "ReplacementChainBroken" and something something, the catalog cannot be used")

Reply:
sounds like really should be reported as an error then.